Signup-Form/src/components/Signup.tsx

```diff
@@ -0,0 +1,272 @@
+    }
+    if (formData.lastName.length == 0) {
+      foundErrors.lastName = "Last Name is required.";
+    } else if (formData.lastName.length < 4 || formData.lastName.length > 25) {
+      foundErrors.lastName = "Last Name should be between 4 and 25 characters.";
+    }
+
+    if (formData.email.length == 0) {
+      foundErrors.email = "Email is required.";
+    } else if (!formData.email.match(emailRegex)) {
+      foundErrors.email = "Email format is invalid.";
+    }
+    if (formData.password.length == 0) {
+      foundErrors.password = "Password is required.";
+    } else if (!formData.password.match(passwordRegex)) {
+      foundErrors.password =
+        "Password must contain minimum 8 characters and at least one special character.";
+    }
+    if (formData.confirmPassword.length == 0) {
+      foundErrors.confirmPassword = "Please confirm password.";
+    } else if (formData.confirmPassword !== formData.password) {
+      foundErrors.confirmPassword = "Password does't match.";
+    }
+    if (formData.country.length == 0) {
+      foundErrors.country = "Please select a country.";
+    }
+    if (formData.city.length == 0) {
+      foundErrors.city = "Please select a city.";
+    }
+    setErrors(foundErrors);
+
+    const isValid = (
+      Object.keys(foundErrors) as [
+        "firstName",
+        "lastName",
+        "email",
+        "password",
+        "confirmPassword",
+        "country",
+        "city"
+      ]
+    ).every((key) => foundErrors[key].length == 0);
+
+    return isValid;
+  };
+
+  const onChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    setFormData({ ...formData, [event.target.name]: event.target.value });
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const isValid = validateValues();
+    if (isValid) {
+      localStorage.setItem("email", formData.email);
+      navigate("/login");
+    }
+  };
+
+  return (
+    <div>
+      <h1>Sign Up</h1>
+      <form id="signupForm" onSubmit={handleSubmit}>
+        <div className="signup-field">
+          <div>
+            <label htmlFor="firstName">First Name</label>
+          </div>
+          <input
+            type="text"
+            name="firstName"
+            id="firstName"
+            onChange={onChange}
+            value={formData.firstName}
+          />
+          {errors.firstName.length !== 0 ? (
+            <div className="error">{errors.firstName}</div>
+          ) : null}
+        </div>
+
+        <div className="signup-field">
+          <div>
+            <label htmlFor="lastName">Last Name</label>
+          </div>
+          <input
+            type="text"
+            name="lastName"
+            id="lastName"
+            onChange={onChange}
+            value={formData.lastName}
+          />
+          {errors.lastName.length !== 0 ? (
+            <div className="error">{errors.lastName}</div>
+          ) : null}
```

Comment:
Varianta mai scurta :D 

```suggestion
          {errors.lastName.length !== 0 && (
            <div className="error">{errors.lastName}</div>
          )}
```